WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -132,11 +132,11 @@ final class ProductSelectorViewModel: ObservableObject {
 
     /// Closure to be invoked when a product is selected or deselected
     ///
-    private let onProductSelectionStateChanged: ((Product) -> Void)?
+    private let onProductSelectionStateChanged: ((Product, Bool) -> Void)?
```

Comment:
General question: What do you think about using a `Bool` to represent "selection state" on these signatures, when the `SelectedState` enum has 3 states and not 2? (.selected, .notSelected, .partiallySelected). Would it be worth to change these to something like the following?
```
private let onProductSelectionStateChanged: ((Product, ProductRow.SelectedState) -> Void)?
```
That said, I'm more inclined to - in favour of simplification- of removing the `.partiallySelected` case, which only seems to modify the style for the parent row when a variation is selected, a row is either selected or unselected.

Reply:
Yeah, I considered that, and it sounds like we came to the same confusion. 

Since we never actually change a product's selection state to `partiallySelected` except as a UI provision. I don't think there's a need to inform the parent view model of that UI-only state, and it makes the code more complex if we do.

If we ever need to track that as an analytic or do something else in the parent view, we can change to use the enum in future.